bin/desi_tile_redshifts

```diff
@@ -0,0 +1,343 @@
+
+Not supported yet: multiple tiles on a single night in a single call:
+
+    desi_tile_redshifts --night 20201215 --tileid 80605 80606 80607
+
+"""
+
+#- Parse command line quickly for --help before slower imports
+import argparse
+p = argparse.ArgumentParser()
+p.add_argument("-n", "--night", type=int, nargs='+', help="YEARMMDD nights")
+p.add_argument("-t", "--tileid", type=int, help="Tile ID")
+p.add_argument("-e", "--expid", type=int, nargs='+', help="exposure IDs")
+p.add_argument("-g", "--group", type=str, help="night group name")
+p.add_argument("--explist", type=str,
+        help="read TILE NIGHT EXPID list from this file")
+p.add_argument("--nosubmit", action="store_true",
+        help="generate scripts but don't submit batch jobs")
+p.add_argument("--batch-queue", type=str, default='realtime',
+        help="batch queue name")
+p.add_argument("--batch-reservation", type=str,
+        help="batch reservation name")
+p.add_argument("--batch-dependency", type=str,
+        help="job dependencies passed to sbatch --dependency")
+
+args = p.parse_args()
+
+import sys, os, glob
+import subprocess
+import numpy as np
+from astropy.table import Table, vstack
+
+from desiutil.log import get_logger
+
+import desispec.io
+from desispec.workflow.tableio import load_table
+from desispec.workflow.exptable import get_exposure_table_pathname
+
+def batch_tile_redshifts(tileid, exptable, group, spectrographs=None,
+    submit=False, queue='realtime', reservation=None, dependency=None):
+    """
+    Generate batch script for spectra+coadd+redshifts for a tile
+
+    Args:
+        tileid (int): Tile ID
+        exptable (Table): has columns NIGHT EXPID to include
+        group (str): group name, e.g. YEARMMDD or "all" or "good"
+
+    Options:
+        spectrographs (list of int): spectrographs to include
+        submit (bool): also submit batch script to queue
+        queue (str): batch queue name
+        reservation (str): batch reservation name
+        dependency (str): passed to sbatch --dependency upon submit
+
+    Returns tuple (scriptpath, error):
+        scriptpath (str): full path to generated script
+        err (int): return code from submitting job (0 if submit=False)
+
+    By default generate the script but don't submit it
+    """
+    log = get_logger()
+    if spectrographs is None:
+        spectrographs = (0,1,2,3,4,5,6,7,8,9)
+
+    spectro_string = ' '.join([str(sp) for sp in spectrographs])
+    num_nodes = len(spectrographs)
+
+    frame_glob = list()
+    for night, expid in zip(exptable['NIGHT'], exptable['EXPID']):
+        frame_glob.append(f'exposures/{night}/{expid:08d}/cframe-[brz]$SPECTRO-{expid:08d}.fits')
+
+    frame_glob = ' '.join(frame_glob)
+
+    reduxdir = desispec.io.specprod_root()
+    scriptdir = f'{reduxdir}/run/scripts/tiles/{tileid}/{group}'
+    os.makedirs(scriptdir, exist_ok=True)
+    batchscript = f'{scriptdir}/coadd-redshifts-{tileid}-{group}.slurm'
```

Comment:
_If_ we want to support single exposures, they break this otherwise valid trend of `/{group}/coadd*-{group}.slurm`, `/{group}/spectra*-{group}.fits`, `/{group}/coadd*-{group}.fits`, `/{group}/zbest*-{group}.fits`, and  ` /{group}/redrock*-{group}.fits`.

I'll leave it up to you whether we want to have each saved under group={expid} and then consolidate all those directories post-factor into an /exposures/ directory. Or whether we want to complicate the logic here to allow for a change in the naming convention